### PR TITLE
Select display currency when sending

### PR DESCRIPTION
**Pull request by zanderz**
Note this uses the displayCurrencies from the current user as the main choices, and the result of SendAssetChoices(from, to) as other choices. Not certain that's right, but SendAssetChoices() doesn't return anything for me.

### CODE REVIEW

**Comment by cjb** [comment]
@zanderz Can you pass the selected display currency to buildPayment so that the "Sending Lumens worth" confirm screen shows that currency?

File: shared/constants/types/wallets.js [hunk 1]
```diff
@@ -176,7 +178,6 @@ export type _State = {
   paymentsMap: I.Map<AccountID, I.List<Payment>>,
   pendingMap: I.Map<AccountID, I.List<Payment>>,
   secretKeyMap: I.Map<AccountID, HiddenString>,
-  selectedAccount: AccountID,
```
Comment (cjb) [inline shared/constants/types/wallets.js]: (Why this change?)

**Comment by cjb** [comment]
@zanderz when I choose a diff currency, the "This is $..." at the bottom of the send screen, and the "Sending Lumens worth" on the confirm screen, don't change away from USD it seems, dunno why

File: shared/constants/types/wallets.js (hunk 1 above)
Comment (zanderz) [reply shared/constants/types/wallets.js]: whoops, that was a mistake, thanks

**Comment by zanderz** [comment]
there were 2 problems: the currency wasn't being added to the buildSend rpc call, but even after adding it, the rpc seems to ignore it and use $ still. Looking into that part.

**Comment by zanderz** [comment]
ok I think this is straightened out - when you pick XLM you can't send that to the service, it has to be null. The only missing thing is the format in the selected currency - it doesn't lead fractional amounts with "0" and @chrisnojima says we shouldn't format it, it should come from the service, but I'm not sure that's available

File: protocol/avdl/stellar1/local.avdl [hunk 2]
```diff
@@ -293,6 +293,8 @@ protocol local {
     // Example: "$1 = 5.0992345 XLM\nSource: coinmarketcap.com"
     string worthInfo;
     array<SendBannerLocal> banners;
+    // Opteinal. Formatted input amount, if local currency is selected.
```
Comment (zanderz) [inline protocol/avdl/stellar1/local.avdl]: whoops, spelling, will change

File: shared/constants/types/wallets.js [hunk 3]
```diff
@@ -83,6 +84,7 @@ export type _BuildingPayment = {
   recipientType: CounterpartyType,
   secretNote: HiddenString,
   to: string,
+  sendAssetChoices: ?Array<StellarRPCTypes.SendAssetChoiceLocal>,
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/constants/types/wallets.js]: done

File: shared/wallets/routes.js [hunk 4]
```diff
@@ -54,6 +55,11 @@ const walletChildren = {
       },
       createNewAccount,
       linkExisting,
+      [Constants.chooseAssetFormRouteKey]: {
```
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/wallets/routes.js]: done

**Comment by zanderz** [comment]
@cjb @buoyad I think I addressed those

File: shared/constants/types/wallets.js [hunk 5]
```diff
@@ -96,6 +98,7 @@ export type _BuiltPayment = {
   toUsername: string,
   worthDescription: string,
   worthInfo: string,
+  amountFormatted: string,
```
Comment (cjb) [inline shared/constants/types/wallets.js]: alphabetize

File: shared/constants/wallets.js [hunk 6]
```diff
@@ -43,6 +45,7 @@ const makeBuiltPayment: I.RecordFactory<Types._BuiltPayment> = I.Record({
   toUsername: '',
   worthDescription: '',
   worthInfo: '',
+  amountFormatted: '',
```
Comment (cjb) [inline shared/constants/wallets.js]: alphabetize

File: shared/constants/wallets.js [hunk 7]
```diff
@@ -84,6 +87,7 @@ const buildPaymentResultToBuiltPayment = (b: RPCTypes.BuildPaymentResLocal) =>
     toUsername: b.toUsername,
     worthDescription: b.worthDescription,
     worthInfo: b.worthInfo,
+    amountFormatted: b.amountFormatted,
```
Comment (cjb) [inline shared/constants/wallets.js]: alphabetize

File: shared/wallets/confirm-form/index.stories.js [hunk 8]
```diff
@@ -31,6 +31,7 @@ const confirmProps = {
   onSendClick: Sb.action('onSendClick'),
   sendFailed: false,
   waiting: false,
+  amountFormatted: '1.234 XLM',
```
Comment (cjb) [inline shared/wallets/confirm-form/index.stories.js]: alphabetize

File: shared/wallets/send-form/asset-input/index.stories.js [hunk 9]
```diff
@@ -15,6 +15,7 @@ const common = {
   onChangeDisplayUnit: Sb.action('onChangeDisplayUnit'),
   topLabel: '',
   value: '',
+  refresh: Sb.action('onRefresh'),
```
Comment (cjb) [inline shared/wallets/send-form/asset-input/index.stories.js]: alphabetize

File: shared/wallets/send-form/choose-asset/index.js [hunk 10]
```diff
@@ -35,11 +36,12 @@ type ExpanderItem = {
   type: 'expander',
 }
 
-type Props = {
+export type Props = {
   displayChoices: Array<DisplayItem>,
+  otherChoices: Array<OtherItem>,
```
Comment (cjb) [inline shared/wallets/send-form/choose-asset/index.js]: alphabetize

File: shared/wallets/send-form/asset-input/index.stories.js (hunk 9 above)
Comment (zanderz) [reply shared/wallets/send-form/asset-input/index.stories.js]: ah missed those, sorry

File: shared/wallets/confirm-form/index.js [hunk 11]
```diff
@@ -21,6 +21,9 @@ type ConfirmSendProps = {|
   banners?: Array<BannerType>,
   sendFailed: boolean,
   waitingKey?: string,
+  symbol?: string,
+  worthDescription?: string,
+  amountFormatted: string,
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/wallets/confirm-form/index.js]: I still don't see this prop used anywhere, could we remove it?

File: shared/wallets/routes.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (zanderz) [reply shared/wallets/routes.js]: done

**Comment by zanderz** [comment]
hoping the the CI failures are unrelated - this is what we're doing, right?

File: protocol/avdl/stellar1/local.avdl [hunk 12]
```diff
@@ -294,6 +294,8 @@ protocol local {
     // Example: "$1 = 5.0992345 XLM\nSource: coinmarketcap.com"
     string worthInfo;
     array<SendBannerLocal> banners;
+    // Optional. Formatted input amount, if local currency is selected.
+    string amountFormatted;
```
Comment (patrickxb) [inline protocol/avdl/stellar1/local.avdl]: Where is this used in the UI?

**Comment by patrickxb** [comment]
I posted in chat, but this doesn't seem to be working unfortunately.  Everything looked fine, but when I sent, it sent the amount as XLM.

I checked the CLI to see if it was a service issue, but the CLI worked fine.

File: protocol/avdl/stellar1/local.avdl (hunk 12 above)
Comment (patrickxb) [reply protocol/avdl/stellar1/local.avdl]: @zanderz ^^^
